pkg/backends/fake/backend.go

```diff
@@ -0,0 +1,161 @@
+	b.config = loadTestV1.ImageDetails{
+		Image: imageName,
+		Tag:   imageTag,
+	}
+}
+
+// SetLogger receives a copy of logger
+func (b *Backend) SetLogger(logger *zap.Logger) {
+	b.logger = logger
+}
+
+// TransformLoadTestSpec use given spec to validate and return a new one or error
+func (*Backend) TransformLoadTestSpec(spec *loadTestV1.LoadTestSpec) error {
+	spec.MasterConfig.Image = imageName
+	spec.MasterConfig.Tag = imageTag
+
+	spec.WorkerConfig.Image = ""
+	spec.WorkerConfig.Tag = ""
+
+	return nil
+}
+
+// SetKubeClientSet receives a copy of kubeClientSet
+func (b *Backend) SetKubeClientSet(kubeClientSet kubernetes.Interface) {
+	b.kubeClient = kubeClientSet
+}
+
+// Sync check if Backend kubernetes resources have been create, if they have not been create them
+func (b *Backend) Sync(ctx context.Context, loadTest loadTestV1.LoadTest, _ string) error {
+	// Get the Namespace resource
+	namespace, err := b.kubeClient.CoreV1().Namespaces().Get(ctx, loadTest.Status.Namespace, metaV1.GetOptions{})
+	// The LoadTest resource may no longer exist, in which case we stop
+	// processing.
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	// Check that we created master job
+	_, err = b.kubeClient.BatchV1().Jobs(namespace.GetName()).Get(ctx, "loadtest-master", metaV1.GetOptions{})
+	if errors.IsNotFound(err) {
+		_, err = b.kubeClient.BatchV1().Jobs(namespace.GetName()).Create(ctx, b.newMasterJob(loadTest), metaV1.CreateOptions{})
+		return err
+	}
+	return err
+}
+
+// SyncStatus check the Backend resources and calculate the current status of the LoadTest from them
+func (b *Backend) SyncStatus(ctx context.Context, _ loadTestV1.LoadTest, loadTestStatus *loadTestV1.LoadTestStatus) error {
+	// Get the Namespace resource
+	namespace, err := b.kubeClient.CoreV1().Namespaces().Get(ctx, loadTestStatus.Namespace, metaV1.GetOptions{})
+	// The LoadTest resource may no longer exist, in which case we stop
+	// processing.
+	if errors.IsNotFound(err) {
+		loadTestStatus.Phase = loadTestV1.LoadTestFinished
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if loadTestStatus.Phase == "" {
+		loadTestStatus.Phase = loadTestV1.LoadTestCreating
+	}
+
+	if loadTestStatus.Phase == loadTestV1.LoadTestErrored {
+		return nil
+	}
+
+	job, err := b.kubeClient.BatchV1().Jobs(namespace.GetName()).Get(ctx, "loadtest-master", metaV1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Get Backend job in namespace and update the LoadTest status with
+	// the Job status
+	loadTestStatus.Phase = getLoadTestPhaseFromJob(job.Status)
+	loadTestStatus.JobStatus = job.Status
+
+	return nil
+}
+
+// newMasterJob creates a new job which runs the Fake master pod
+func (b *Backend) newMasterJob(loadTest loadTestV1.LoadTest) *batchV1.Job {
+	imageRef := fmt.Sprintf("%s:%s", loadTest.Spec.MasterConfig.Image, loadTest.Spec.MasterConfig.Tag)
+	if "" == loadTest.Spec.MasterConfig.Image || "" == loadTest.Spec.MasterConfig.Tag {
```

Comment:
this is C-style =) this type of error (missing of one `=`) is not possible in golang `if` =)